src/SepaController.php

```diff
@@ -59,14 +59,21 @@ protected function fetchIntent(\Payment $payment, Api $api, array $expand = [])
     if ($recurring) {
       $expand[] = 'mandate';
       $expand[] = 'payment_method';
-    }
-    $intent = parent::fetchIntent($payment, $api, $expand);
-    if ($recurring) {
+      $intent = parent::fetchIntent($payment, $api, $expand);
       $payment->stripe_sepa = [
         'mandate_reference' => $intent['mandate']['payment_method_details']['sepa_debit']['reference'],
         'last4' => $intent['payment_method']['sepa_debit']['last4'],
       ];
     }
+    else {
+      $intent = parent::fetchIntent($payment, $api, $expand);
+      $details = $intent['charges']['data'][0]['payment_method_details']['sepa_debit'];
+      $mandate = $api->retrieveMandate($details['mandate']);
+      $payment->stripe_sepa = [
```

Comment:
I think I'd find it a bit easier to read (or expand in the future) when `$payment->stripe_sepa` was set outside the if? Then for recurring payments, we'd also need to set
```php
$details = $intent['payment_method']['sepa_debit'];
$mandate = $intent['mandate'];
```

Whatever version you decide on, please add a comment to one-off $details explaining the detour using charges is necessary because neither PaymentIntent nor PaymentMethod include a mandate (at the moment).

Reply:
Ok, thanks will do that …